### PR TITLE
DictType: added support for int keys

### LINKED ISSUE #297: keys of `int` type are not supported for dicts
int type keys get coerced to str type and cause issues with type checks

### PR DESCRIPTION
**Pull request by maheshambule**
adds support for keys of int types in dicts
currently int keys were coerced to str type 

fixes https://github.com/iterative/mlem/issues/297

### CODE REVIEW

File: tests/core/test_data_type.py [hunk 1]
```diff
@@ -246,3 +254,64 @@ def custom_assert(x, y):
...
+    payload2["item_types"]["3"]["item_types"][1] = payload2["item_types"]["3"][
+        "item_types"
+    ].pop("1")
+    return payload2
+
+
+@pytest.mark.parametrize(
+    "d_value,payload",
+    [
+        ({"1": 1.5, "2": "a", "3": {"1": False}}, payload1()),
```
Comment (mike0sv) [inline tests/core/test_data_type.py:285]: make payload fixtures and use `lazy_fixture`?

**Review by aguschin — APPROVED** [review]
LGTM! Congrats on finalizing this PR @maheshambule! Would you like to merge this yourself? :)

**Comment by aguschin** [comment]
@maheshambule
srry, do you mind merging this?)

**Comment by aguschin** [comment]
Errrr... Looks like a strange merge conflict to me. Will investigate tomorrow.

**Comment by aguschin** [comment]
Great! Congrats @maheshambule :)

**Pull request merged**